src/hotspot/share/gc/shared/memAllocator.hpp

```diff
@@ -118,4 +118,44 @@ class ClassAllocator: public MemAllocator {
   virtual oop initialize(HeapWord* mem) const;
 };
 
+// Manages a scope where a failed heap allocation results in
```

Comment:
I think you need to document why disable_events exists as well.

Reply:
I've removed `disable_events` altogether.